### ISSUE TITLE
Api.list!/2 doesn't work with the Postgres Data Layer

### ISSUE THREAD

**Issue by rbino** [ash-project/ash_postgres#225]
**Describe the bug**
I've just updated a project from ~3 months ago and the call to `Api.list!(Resource, :id)` is crashing.

**To Reproduce**
Here's a failing test (I've tried it in `test/select_test.exs`):

```elixir
  test "list! works" do
    %{id: id} =
      Post
      |> Ash.Changeset.new(%{title: "title"})
      |> Api.create!()

    assert [^id] = Api.list!(Post, :id)
  end
```

**Expected behavior**
I'd expect `list!` to return the requested list of fields.

** Runtime
 - Elixir version: 1.16.2
 - Erlang version: 26.2.2
 - OS: Debian Testing
 - Ash version: 2.20.3
 - any related extension versions: ash_postgres from `main`

**Additional context**
I'm opening this in `ash_postgres` because I've tried to add a similar test in the Ash suite (with the ETS data layer) and it doesn't fail there.

**Comment by rbino** [comment]
Here's the stacktrace from the failing test:

```
  1) test list! works (AshPostgres.SelectTest)
     test/select_test.exs:8
     ** (KeyError) key :sort not found in: nil

     If you are using the dot syntax, such as map.field, make sure the left-hand side of the dot is a map
     code: assert [^id] = Api.list!(Post, :id)
     stacktrace:
       (ash_postgres 1.5.21) lib/aggregate.ex:1201: AshPostgres.Aggregate.add_subquery_aggregate_select/6
       (ash_postgres 1.5.21) lib/data_layer.ex:941: anonymous fn/4 in AshPostgres.DataLayer.add_single_aggs/4
       (elixir 1.16.2) lib/enum.ex:2528: Enum."-reduce/3-lists^foldl/2-0-"/3
       (ash_postgres 1.5.21) lib/data_layer.ex:844: AshPostgres.DataLayer.run_aggregate_query/3
       (ash 2.20.3) lib/ash/actions/aggregate.ex:77: anonymous fn/5 in Ash.Actions.Aggregate.run/4
       (elixir 1.16.2) lib/enum.ex:4839: Enumerable.List.reduce/3
       (elixir 1.16.2) lib/enum.ex:2582: Enum.reduce_while/3
       (ash_postgres 1.5.21) test/support/api.ex:1: AshPostgres.Test.Api.list!/3
       test/select_test.exs:14: (test)
 ```

**Comment by zachdaniel** [comment]
Fixed in 1.5.22
